azureproject/settings.py

```diff
@@ -93,3 +93,67 @@
 # https://docs.djangoproject.com/en/4.0/ref/settings/#default-auto-field
 
 DEFAULT_AUTO_FIELD = 'django.db.models.BigAutoField'
+
+if 'RUNNING_IN_PRODUCTION' in os.environ:
+    # Configure allowed host names that can be served and trusted origins for Azure Container Apps.
+    ALLOWED_HOSTS = ['.azurecontainerapps.io']
+    CSRF_TRUSTED_ORIGINS = ['https://*.azurecontainerapps.io']
+    DEBUG = False
+    DEBUG_PROPAGATE_EXCEPTIONS = True
+
+    # SECURITY WARNING: keep the secret key used in production secret!
+    # Use this py command to create secret 
+    # python -c 'import secrets; print(secrets.token_hex())'
+    SECRET_KEY = os.getenv('AZURE_SECRET_KEY')
+
+    STATICFILES_STORAGE = 'whitenoise.storage.CompressedStaticFilesStorage'
+
+    # Configure database connection for Azure PostgreSQL Flexible server instance.
+    # AZURE_POSTGRESQL_HOST is the full URL.
+    # AZURE_POSTGRESQL_USERNAME is just name without @server-name.
+    DATABASES = {
+        'default': {
+            'ENGINE': 'django.db.backends.postgresql',
+            'NAME': os.environ['AZURE_POSTGRESQL_DATABASE'],
+            'HOST': os.environ['AZURE_POSTGRESQL_HOST'],
+            'USER': os.environ['AZURE_POSTGRESQL_USERNAME'],
+            'PASSWORD': os.environ['AZURE_POSTGRESQL_PASSWORD'], 
+        }
+    }
+else:
+    # SECURITY WARNING: don't run with debug turned on in production!
+    DEBUG = True
+
+    # Use this py command to create secret 
+    # python -c 'import secrets; print(secrets.token_hex())'
+    SECRET_KEY = os.getenv('LOCAL_SECRET_KEY')
+
+    # Don't use Whitenoise to avoid having to run collectstatic first.
+    STATICFILES_STORAGE = 'django.contrib.staticfiles.storage.StaticFilesStorage'
+
+    ALLOWED_HOSTS = ['*']
+
+    # Configure connection setting for PostgreSQL instance.
+    # Set these environment variables in the .env file for this project.  
+    if 'USE_REMOTE_POSTGRESQL' in os.environ:
```

Comment:
Eek, are customers asking for this? I stay far away from prod databases when I'm local, unless I have a good way to explicitly turn on read-only access to them.

Reply:
@pamelafox isn't this a valid debugging scenario? i.e. running locally against a DB hosted in Azure to flush out any bugs that might arise due to implementation differences between the cloud and local implementation of the DB? It doesn't necessarily have to be the production DB.